tests/meshcop/commissioner.cpp

```diff
@@ -241,47 +165,90 @@ void HandleRelayReceive(const Coap::Resource &aResource, const Coap::Message &aM
     return;
 }
 
+/* from the example:
+ * http://beej.us/guide/bgnet/output/html/multipage/inet_ntopman.html
+ */
+static char *get_ip_str(const struct sockaddr *sa, char *s, size_t maxlen)
+{
+    switch (sa->sa_family)
+    {
+    case AF_INET:
+        inet_ntop(AF_INET, &(((struct sockaddr_in *)sa)->sin_addr),
+                  s, maxlen);
+        break;
+
+    case AF_INET6:
+        inet_ntop(AF_INET6, &(((struct sockaddr_in6 *)sa)->sin6_addr),
+                  s, maxlen);
+        break;
+
+    default:
+        strncpy(s, "Unknown AF", maxlen);
+        return NULL;
+    }
+
+    return s;
+}
+
 int SendRelayTransmit(Context &aContext)
 {
-    uint8_t payload[kSizeMaxPacket];
-    uint8_t dtlsEncapsulation[kSizeMaxPacket];
-    Tlv    *responseTlv = reinterpret_cast<Tlv *>(payload);
+    uint8_t            payload[kSizeMaxPacket];
+    uint8_t            dtlsEncapsulation[kSizeMaxPacket];
+    Tlv               *responseTlv = reinterpret_cast<Tlv *>(payload);
+    struct sockaddr_in from_addr;
+    socklen_t          addrlen;
+
+    addrlen = sizeof(from_addr);
+
 
-    ssize_t ret = recvfrom(aContext.mSocket, dtlsEncapsulation, sizeof(dtlsEncapsulation), 0, NULL, NULL);
+    ssize_t ret =
+        recvfrom(aContext.mSocket, dtlsEncapsulation, sizeof(dtlsEncapsulation), 0, (struct sockaddr *)(&from_addr),
+                 &addrlen);
 
     VerifyOrExit(ret > 0);
+    {
+        // Print this, because in some environments...
+        // other things on the network use the same port
+        // as open thread is using here :-(
+        char buf[50];
+        get_ip_str((struct sockaddr *)(&from_addr), buf, sizeof(buf));
```

Comment:
No - we use ipv4 and we use ipv6 - the android app does both, hence it does both.